app/controllers/api/v2/smart_variables_controller.rb

```diff
@@ -44,7 +44,7 @@ def show
       def create
         @smart_variable   = LookupKey.new(params[:smart_variable]) unless @puppetclass
         @smart_variable ||= @puppetclass.lookup_keys.build(params[:smart_variable])
-        @smart_variable.save!
+        process_response @smart_variable.save!
```

Comment:
Think you need to remove the `!` here so it returns true/false instead of an exception on failure.